### DISCUSSION TITLE
Make line-too-long(E501) ignore deep dives into dictionaries

### DISCUSSION THREAD

**Discussion by Tosh0kan** [astral-sh/ruff#8875] [Ideas]
In my code, there's this snippet:
```py

                                if title.json()["data"]["id"] == feed.json()["data"][0]["relationships"][1]["id"]:
                                    init_states.append({title.json()['data']['attributes']["title"]["en"]:
                                        feed.json()["data"][0]})
                                    title_urls.remove(str(title.url))
                                    feed_urls.remove(str(feed.url))
                                    break
                                else:
                                    continue
```
As you can see, I'm working with a dictionary that has many, many, many, _many_ layers, and doing comparisons between different elements of that same dictionary (in reality, a JSON file got through a GET request) so I get a lot of nested indexes on both sides of the comparison operation, and there's no neat to condense it, and since these aren't inside parenthesis, I can't separate them by line, so if you could implement a check for this kind of situation, it would be awesome.

**Comment by charliermarsh** [comment]
I can see why this is inconvenient, but I think for me this falls pretty squarely under the intent of the E501 rule. In other words, I think folks would find it _surprising_ if these were excluded, since those are overly long lines that could be further broken-down (for example, could you assign `title.join()['data']` to a variable, and access that instead?).

**Comment by Tosh0kan** [comment]
Oh yeah, definitely. In fact, searching around before starting this discussion, [doing something similar to that ](https://stackoverflow.com/a/52187020) seems to be the preferred way from getting around it. The thing is, in my opinion, doing that would result in the creation of otherwise unnecessary variables, nesting, and complexity, which goes against certain parts of the zen of python, which spawned PEP8.
